frontend/src/model/beta/overview/ModelCardHistoryDialog.tsx

```diff
@@ -0,0 +1,63 @@
+import { Button, DialogActions, DialogTitle, Paper, Table, TableContainer } from '@mui/material'
+import Dialog from '@mui/material/Dialog'
+import { ThemeProvider, useTheme } from '@mui/material/styles'
+import { useMemo } from 'react'
+
+import { useModelCardRevisions } from '../../../../actions/modelCard'
+import Loading from '../../../../src/common/Loading'
+import MessageAlert from '../../../../src/MessageAlert'
+import { ModelInterface } from '../../../../types/v2/types'
+import { sortByCreatedAtDescending } from '../../../../utils/dateUtils'
+import ModelCardHistoryHeaderTable from '../overview/ModelCardHistoryTable'
+import ModelCardRevisionListDisplay from '../overview/ModelCardRevisionListDisplay'
+
+type DialogProps = {
+  model: ModelInterface
+  open: boolean
+  setOpen: (isOpen: boolean) => void
+}
+
+export default function ViewModelCardHistoryDialog({
+  model,
+
+  open,
+  setOpen,
+}: DialogProps) {
+  const theme = useTheme()
+  const { modelCardRevisions, isModelCardRevisionsLoading, isModelCardRevisionsError } = useModelCardRevisions(model.id)
+  const sortedModelCardRevisions = useMemo(
+    () => modelCardRevisions.sort(sortByCreatedAtDescending),
+    [modelCardRevisions],
+  )
+
+  if (isModelCardRevisionsError) {
+    return <MessageAlert message={isModelCardRevisionsError.info.message} severity='error' />
+  }
+
+  return (
+    <>
+      {isModelCardRevisionsLoading && <Loading />}
+      <ThemeProvider theme={theme}>
```

Comment:
We don't need to use this provider as it's already provided at the top level of the application. Calling `theme` will still work within your JSX.